ports/esp32/machine_uart.c

```diff
@@ -243,6 +251,59 @@ STATIC void machine_uart_init_helper(machine_uart_obj_t *self, size_t n_args, co
     }
     self->invert = args[ARG_invert].u_int;
     uart_set_line_inverse(self->uart_num, self->invert);
+
+    // set mode
+    switch (args[ARG_uart_mode].u_int) {
```

Comment:
This switch should be eliminated by providing `machine.UART.xxx` constants that have the right value to pass to uart_set_mode and assignment to `uart_mode`.